src/olympia/reviewers/management/commands/auto_reject.py

```diff
@@ -0,0 +1,121 @@
+            default=False,
+            help='Fetch version candidates and perform all checks but do not '
+                 'actually reject anything.')
+
+    def fetch_addon_candidates(self, *, now):
+        """Return a queryset with the Addon instances that have versions that
+        should be considered for rejection (deadline before 'now')."""
+        return (
+            Addon.unfiltered
+                 .filter(versions__reviewerflags__pending_rejection__lt=now)
+                 .order_by('id').distinct()
+        )
+
+    def fetch_version_candidates_for_addon(self, *, addon, now):
+        """Return a queryset with the versions that should be considered for
+        rejection (deadline before 'now') for a given add-on."""
+        return (
+            addon.versions(manager='unfiltered_for_relations')
+                 .filter(reviewerflags__pending_rejection__lt=now)
+                 .order_by('id')
+        )
+
+    @transaction.atomic
+    def reject_versions(self, *, addon, versions, latest_version):
+        """Reject specific versions for an addon."""
+        if self.dry_run:
+            log.info('Would reject versions %s from add-on %s but this is a '
+                     'dry run.', versions, addon)
+            return
+        helper = ReviewHelper(addon=addon, version=latest_version)
+        helper.handler.data = {
+            'comments': 'Automatic rejection after grace period ended.',
+            'versions': versions,
+        }
+        helper.handler.reject_multiple_versions()
+        VersionReviewerFlags.objects.filter(
+            version__in=list(versions)).update(pending_rejection=None)
+
+    def process_addon(self, *, addon, now):
+        latest_version = addon.find_latest_version(
+            channel=amo.RELEASE_CHANNEL_LISTED)
+        if (latest_version and latest_version.is_unreviewed and
+                not latest_version.pending_rejection):
+            # If latest version is unreviewed and not pending
+            # rejection, we want to put the delayed rejection of all
+            # versions of this addon on hold until a decision has been
+            # made by reviewers on the latest one.
+            log.info(
+                'Skipping rejections for add-on %s until version %s '
+                'has been reviewed', addon.pk, latest_version.pk)
+            return
+        versions = self.fetch_version_candidates_for_addon(
+            addon=addon, now=now)
+        if not versions.exists():
+            log.info(
+                'Somehow no versions to auto-reject for add-on %s', addon.pk)
+            return
+        locked_by = get_reviewing_cache(addon.pk)
+        if locked_by:
+            # Don't auto-reject something that has been locked, even by the
+            # task user - wait until it's free to avoid any conflicts.
+            log.info(
+                'Skipping rejections for add-on %s until lock from %s '
+                'has expired', addon.pk, locked_by)
```

Comment:
`locked_by` returns a numeric value for me (like `1`), maybe we could rephrase this log statement like: `...until lock from reviewer %s has expired` or something.

Reply:
Yeah it's an user id. I adjusted the message in https://github.com/mozilla/addons-server/pull/14838/commits/e2c8b3a7c1494d9b963c69e0256e3797ae5309f2